### PR TITLE
#482 new composer plugin

### PR DESCRIPTION
**Pull request by kaustabhbarman**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -51,8 +51,12 @@
   },
   "minimum-stability": "dev",
   "prefer-stable": true,
-  "config": {
-    "component-dir": "public/assets"
+
+  "extra": {
+    "installer-types": ["component", "component"],
```
Comment (j3nsch) [inline composer.json:56]: Is it really necessary to write this twice? Please add a link to the documentation for this to the issue.

File: composer.json [hunk 2]
```diff
@@ -27,10 +27,10 @@
...
     "opus4-repo/search": "dev-master",
     "opus4-repo/opus4-bibtex": "dev-master",
-    "components/jquery": "3.4.*",
-    "components/jqueryui": "1.12.*",
-    "robloach/component-installer": "*",
-    "symfony/console": "*"
+    "components/jquery": "*",
+    "components/jqueryui": "*",
+    "symfony/console": "*",
+    "oomphinc/composer-installers-extender": "^2.0"
```
Comment (j3nsch) [inline composer.json:33]: Please move it in the old position, so components and installer are next to each other.

**Review by j3nsch — CHANGES_REQUESTED** [review]
Are there any other changes with the new installer? What does "allow-plugins" do? What are the consequences? Anything relevant should go into the issue. Thanks!

**Comment by kaustabhbarman** [comment]
> Are there any other changes with the new installer? What does "allow-plugins" do? What are the consequences? Anything relevant should go into the issue. Thanks!

As part of a new security feature, Composer 2 requires that plugins that have executable code need to be explicitly allowed in `composer.json` when installing it for the first time. A user input "yes" adds this "allow-plugins" config to `composer.json`. I faced it even when setting up OPUS 4 (master or 4.7.1 branch) recently with Composer 2. So I can remove it for now in this PR, but it might show up again later from anyone who has set up OPUS with composer 2 in the future.

File: composer.json (hunk 2 above)
Comment (kaustabhbarman) [reply composer.json:33]: Sure.

File: composer.json (hunk 1 above)
Comment (kaustabhbarman) [reply composer.json:56]: No. Thought it might be necessary, but it isn't. Will remove it.

File: composer.json [hunk 3]
```diff
@@ -27,9 +27,9 @@
     "opus4-repo/framework": "dev-master",
     "opus4-repo/search": "dev-master",
     "opus4-repo/opus4-bibtex": "dev-master",
-    "components/jquery": "3.4.*",
-    "components/jqueryui": "1.12.*",
-    "robloach/component-installer": "*",
+    "components/jquery": "*",
```
Comment (j3nsch) [inline composer.json:30]: You cannot simply remove version numbers without documenting it. The versions are there for a reason. Probably the newest versions will be fine right now, we can test that, but perhaps not anymore in a few month, next year.

Comment (j3nsch) [reply composer.json:30]: I am putting the old versions back in.

**Pull request merged**